lib/cli_common/cli_common/mercurial.py

```diff
@@ -5,8 +5,21 @@
 
 import hglib
 
-MOZILLA_CENTRAL = 'https://hg.mozilla.org/mozilla-central'
-MOZILLA_UNIFIED = 'https://hg.mozilla.org/mozilla-unified',
+import cli_common.log
+
+log = cli_common.log.get_logger(__name__)
+
+
+def hg_run(cmd):
+    '''
+    Run a mercurial command without an hglib instance
+    Useful for initial custom clones
+    '''
+    proc = hglib.util.popen([hglib.HGPATH] + cmd)
+    out, err = proc.communicate()
+    if proc.returncode:
+        raise hglib.error.CommandError(cmd, proc.returncode, out, err)
+    return out
 
 
 def robust_checkout(repo_url, repo_dir, branch=b'tip', revision=None):
```

Comment:
BTW, we should probably use this everywhere (in SA, CCOV and bugbug)

Reply:
robust or batch ?

Reply:
I meant robust, it seems like we are duplicating this code in all projects

Reply:
Well, all projects except pulselistener :P